File: app/src/main/kotlin/org/coffeetrain/daggerkotlintestfail/MyComponent.java
package org.coffeetrain.daggerkotlintestfail;

import dagger.Component;
import javax.inject.Singleton;
import org.coffeetrain.daggerkotlintestfail.MainActivity.Taco;

@Singleton
@Component(modules = { MyModule.class })
public interface MyComponent {
  void inject(MyApp myApp);

  void inject(MyApp.Bar bar);

  void inject(Taco taco);
}

File: app/src/main/kotlin/org/coffeetrain/daggerkotlintestfail/MyComponent.kt
package org.coffeetrain.daggerkotlintestfail

import dagger.Component
import org.coffeetrain.daggerkotlintestfail.MainActivity.Taco
import javax.inject.Singleton

@Singleton
@Component(modules = [MyModule::class])
interface MyComponent {
  fun inject(myApp: MyApp)
  fun inject(myAppBar: MyApp.Bar)
  fun inject(taco: Taco)
}
